packages/dd-trace/src/exporters/agent/writer.js

```diff
@@ -42,7 +42,7 @@ class Writer extends BaseWriter {
       startupLog({ agentError: err })
 
       if (err) {
-        log.error('Error sending payload to the agent (status code: %s)', err.status, err)
+        log.error('Error sending payload to the agent (status code: %s)', err.status, err, log.NO_TELEMETRY)
```

Comment:
Awesome thanks!

Reply:
This pushes knowledge of telemetry reporting (or any logging subscriber in general) to the logging client. I'd prefer if it can be handled downstream, and if not, a more generic flag would make more sense, for example `log.SENSITIVE` which subscribers can then use to decide whether to report sensitive data.